### ISSUE TITLE
Getting Failed to initialize suite datastore. 

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge-java#100]
**Expected behavior**
should be able to run without an exception

**Actual behavior**
Getting `Failed to initialize suite datastore. Error: write tcp 127.0.0.1:50260->127.0.0.1:50264: use of closed network connection`

**Steps to replicate**
* Project [Gradle example](https://github.com/manupsunny/gauge-gradle-sample)
* run `./gradlew gauge`
with gradle 2.5 it works fine
* change the gradle-wrapper.properties to use gradle 4.0
And get the exception
```
Exception in thread "main" org.reflections.ReflectionsException: Can't resolve member named helloWorld for class StepImplementation
    at org.reflections.util.Utils.getMemberFromDescriptor(Utils.java:94)
    at org.reflections.util.Utils.getMethodsFromDescriptors(Utils.java:101)
    at org.reflections.Reflections.getMethodsAnnotatedWith(Reflections.java:482)
    at com.thoughtworks.gauge.scan.StepsScanner.scan(StepsScanner.java:38)
    at com.thoughtworks.gauge.scan.ClasspathScanner.scan(ClasspathScanner.java:39)
    at com.thoughtworks.gauge.GaugeRuntime.connectSynchronously(GaugeRuntime.java:81)
    at com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:53)
process /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre/bin/java with pid 6836 quit unexpectedly. exit status 1
[ValidationError] /Users/sswaroop/work/gauge-gradle-sample/sub-project/subSpecs/sample.spec:10 Connection closed [127.0.0.1:50264] cause: EOF => 'A context step which gets executed before every scenario'
[ValidationError] /Users/sswaroop/work/gauge-gradle-sample/sub-project/subSpecs/sample.spec:17 write tcp 127.0.0.1:50260->127.0.0.1:50264: use of closed network connection => 'Say "hello (from sub)" to "gauge"'
[ValidationError] /Users/sswaroop/work/gauge-gradle-sample/sub-project/subSpecs/sample.spec:25 write tcp 127.0.0.1:50260->127.0.0.1:50264: use of closed network connection => 'Say "hello again (from sub)" to "gauge"'
[ValidationError] /Users/sswaroop/work/gauge-gradle-sample/sub-project/subSpecs/sample.spec:26 write tcp 127.0.0.1:50260->127.0.0.1:50264: use of closed network connection => 'Step that takes a table <table>'
  Failed to initialize suite datastore. Error: write tcp 127.0.0.1:50260->127.0.0.1:50264: use of closed network connection
```

**Version**
```
Gauge version: 0.9.0.nightly-2017-07-07

Plugins
-------
csharp (0.10.1)
html-report (4.0.1)
java (0.6.4)
spectacle (0.1.2)
```

**Comment by cesarloachamin** [comment]
I have the same problem with gradle v4.5.0.
```
Gauge version: 0.9.7
Plugins
-------
html-report (4.0.3)
java (0.6.6)
screenshot (0.0.1)
spectacle (0.1.2)
```

**Comment by riju91** [comment]
@cesarloachamin  Please have a look at this issue https://github.com/manupsunny/gauge-gradle-plugin/issues/20

**Comment by cesarloachamin** [comment]
Hey @riju91 I checked the issue you mention, and it's similar to the one reported here. This is the stack trace
```
Exception in thread "main" org.reflections.ReflectionsException: Can't resolve member named openProductEditPage for class com.thoughtworks.gauge.example.pages.ProductSpec
    at org.reflections.util.Utils.getMemberFromDescriptor(Utils.java:94)
    at org.reflections.util.Utils.getMethodsFromDescriptors(Utils.java:101)
    at org.reflections.Reflections.getMethodsAnnotatedWith(Reflections.java:482)
    at com.thoughtworks.gauge.scan.StepsScanner.scan(StepsScanner.java:38)
    at com.thoughtworks.gauge.scan.ClasspathScanner.scan(ClasspathScanner.java:39)
    at com.thoughtworks.gauge.GaugeRuntime.connectSynchronously(GaugeRuntime.java:82)
    at com.thoughtworks.gauge.GaugeRuntime.main(GaugeRuntime.java:54)
process /usr/bin/java with pid 2026 quit unexpectedly. exit status 1
[ValidationError] /Users/cesar/Development/factoware/learn/acceptance-tests/specs/customer.spec:7 Connection closed [127.0.0.1:50598] cause: read tcp 127.0.0.1:50595->127.0.0.1:50598: read: connection reset by peer => 'On the customer page'
[ValidationError] /Users/cesar/Development/factoware/learn/acceptance-tests/specs/customer.spec:15 write tcp 127.0.0.1:50595->127.0.0.1:50598: use of closed network connection => 'Search for customer "ScroogeMcduck"'
  Failed to initialize suite datastore. Error: write tcp 127.0.0.1:50595->127.0.0.1:50598: use of closed network connection
Successfully generated html-report to => /Users/cesar/Development/factoware/learn/acceptance-tests/reports/html-report/index.html

```

**Comment by zabil** [comment]
Closing this as an old issue. This may be fixed in the newer releases. 
Will reopen if it still needs fixing.
